### PR TITLE
generic ida play pattern

### PR DESCRIPTION
**Pull request by macdiesel**
@edx/pipeline-team PTAL.

### CODE REVIEW

File: edxpipelines/patterns/tasks.py [hunk 1]
```diff
@@ -377,39 +380,38 @@ def generate_run_app_playbook(job, secure_dir, playbook_path, runif="passed"):
         secure_dir (str): name of dir containing the edx-ops/configuration-secure repo
         playbook_path (str): path to playbook relative to the top-level 'configuration' directory
         runif (str): one of ['passed', 'failed', 'any'] Default: passed
+        extra_args (dict): a dictionary of extra arguments that should be passed to the ansible play
```
Comment (feanil) [inline edxpipelines/patterns/tasks.py]: This should be **kwargs.  So it looks like you considered not using kwargs and just passing a specific dictionary for the ansible vars.  Why did you end up going with kwargs?

Comment (feanil) [inline edxpipelines/patterns/tasks.py]: Oh I see, it's much cleaner to call theses from your pipelines if you do it this way.  The one concern I have is that this means that some arguments to the function become ansible variables while others don't which is something that I could see the end user of the function getting confused by, requiring them to look at this code/function definition to figure it out.

Comment (macdiesel) [inline edxpipelines/patterns/tasks.py]: My thinking on this was more along the lines of: All of these variables are needed to execute a play, the major ones are defined in the method signature, anything else extra that you pass in gets sent as a -e option. I think that for the caller of the function it makes it a little cleaner for them to pass down their options for the playbook.

This is mostly an attempt to make this more like a DSL.

Also I would argue this is python, need to call something? You usually take a look at the method you are calling and the docs.

Comment (feanil) [inline edxpipelines/patterns/tasks.py]: seems reasonable.

**Comment by feanil** [comment]
:+1:  once the comment is fixed to talk about kwargs instead of extra_args.

**Pull request merged**